### PR TITLE
Avahi service registering

### PR DESCRIPTION
**Pull request by rmescandon**
This is the same solution as snapweb uses. It publishes current hostname in local dns, in this case, every time any of the http servers starts up.
Portals should be accesible by:

http://[hostname].local:8080

### CODE REVIEW

File: daemon/daemon.go [hunk 1]
```diff
@@ -161,6 +162,8 @@ func main() {
 			managementServerDown()
 			fmt.Println("==== Start Operational Mode http server if not running")
 			operationalServerUp()
+			//TODO POC avahi
+			avahi.InitMDNS()
```
Comment (knitzsche) [inline daemon/daemon.go]: for now the operational server code is commented out, so please rebase on master.

File: avahi/avahi.go [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+type mdnsScanner interface {
+	ScanInterfaces() (string, error)
+}
+
+var _mdns mdnsScanner
+
+var initOnce sync.Once
+
+const hostnameDefault = "snapweb"
```
Comment (knitzsche) [inline avahi/avahi.go]: not sure "snapweb" is the right default hostname to use when it cannot be obtained.

File: daemon/daemon.go (hunk 1 above)
Comment (rmescandon) [inline daemon/daemon.go]: done

File: avahi/avahi.go (hunk 2 above)
Comment (rmescandon) [inline avahi/avahi.go]: haha, indeed. The problems of copy and paste. I've set "device" as the default one.

**Comment by rmescandon** [comment]
addressed comments and merged trunk

File: daemon/daemon.go [hunk 3]
```diff
@@ -217,5 +219,7 @@ func main() {
 
 		fmt.Println("==== Start Management portal if not running")
 		managementServerUp()
+		//TODO POC avahi
+		avahi.InitMDNS()
 	}
```
Comment (knitzsche) [inline daemon/daemon.go]: should this be moved into the managementServerUp() method, particular inside the code block that only runs when the management server is not already up: https://github.com/CanonicalLtd/UCWifiConnect/blob/master/daemon/daemon.go#L99 ?

**Pull request merged**